forum/qa-plugin/report-reason/q2apro-flag-reasons-validation.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+class q2apro_flag_reasons_validation {
+    protected function wrapFlagReasonError($errorCode) {
+        return ['processingFlagReasonError' => $errorCode];
+    }
+
+    protected function handleReportError($errorCode) {
+        $wrappedErrorCode = $this->wrapFlagReasonError($errorCode);
+        echo json_encode($wrappedErrorCode);
+
+        $wrappedErrorCode['_reportReasonRequestJSON'] = $this->_reportReasonRequestJSON;
+        error_log(json_encode($wrappedErrorCode));
+    }
+
+    protected function outputErrorToResponseAndExit($errorCode) {
+        $this->handleReportError(errorCode);
+        exit();
+    }
+
+    protected function isValidJSON() {
+        if (json_last_error() != JSON_ERROR_NONE) {
+            $this->handleReportError('REQUEST_IS_NOT_VALID_JSON');
+            return false;
+        }
+
+        return true;
+    }
+
+    protected function isDataSet($data) {
+        if (empty($data)) {
+            $this->handleReportError('EMPTY_REQUEST');
+            return false;
+        }
+
+        return true;
+    }
+
+    protected function isValidData($props) {
+        $propsModel = [
+            'notice' => '',
+            'postId' => 0,
+            'postType' => '',
+            'questionId' => 0,
+            'reasonId' => 0,
+            'reportType' => 'addFlag'
+        ];
+        $propsKeys = array_keys($props);
+        $matchedPropsKeysNum = 0;
+        $noticeProp = 'notice';
+        $reportTypeProp = 'reportType';
+
+//            var_dump('$props: ' ,  $props);
+
+        foreach ($propsModel as $key => $value) {
+            $propValue = @$props[$key];
+//                echo('$key: '. $key. ' /$propValue: '. $propValue. ' /$propsModel[$key]: '.$propsModel[$key]);
+
+            /*
+                TODO:
+                    -   Should validate against request URL referrer property?
+                        To check if i.e questionId request property is contained by page URL.
+                    -   Should validate if specified post (question, answer or comment) is already flagged by the User?
+                    -   Should validate post security 'code' contained by <input [hidden]> on page?
```

Comment:
znów wykomentowany kod i zapiski z którymi nie za bardzo wiem co mam zrobić

Reply:
Zapisy TODO w kodzie zostawiłem specjalnie, aby pamiętać co i gdzie trzeba jeszcze poprawić przed zatwierdzeniem PRa. Część z nich opisałem w treści PRa.

W tym przypadku to są pytania, na które spodziewałem się uzyskać odpowiedź właśnie podczas Code Review. ;)